src/btree/bt_split.c

```diff
@@ -340,9 +340,18 @@ __split_ref_move(WT_SESSION_IMPL *session, WT_PAGE *from_home,
 			return (ret);
 		}
 		addr->size = (uint8_t)unpack.size;
-		addr->type =
-		    unpack.raw == WT_CELL_ADDR_INT ? WT_ADDR_INT : WT_ADDR_LEAF;
-		ref->addr = addr;
```

Comment:
I think this line is need? I tested against the truncate workload for WT-2240 and it started segfaulting. Adding this line back seems to resolve.